src/site_upload/check_if_complete/check_if_complete.py

```diff
@@ -0,0 +1,140 @@
+import datetime
+import json
+import logging
+import os
+from time import sleep
+
+import awswrangler
+import boto3
+import botocore
+
+from shared import enums, functions
+
+logger = logging.getLogger()
+logger.setLevel("INFO")
+s3_client = boto3.client("s3")
+sns_client = boto3.client("sns")
+g_client = boto3.client("glue")
+
+
+def check_if_complete(message) -> (bool, dict):
+    transaction = functions.get_s3_json_as_dict(
+        bucket=os.environ.get("BUCKET_NAME"),
+        key=(
+            f"{enums.BucketPath.META.value}/transactions/{message['site']}__{message['study']}.json"
+        ),
+        s3_client=s3_client,
+    )
+    upload_time = datetime.datetime.fromisoformat(transaction["uploaded_at"])
+    for filetype, source, suffix in [
+        ("cube", enums.BucketPath.AGGREGATE.value, "aggregate"),
+        ("annotated_cube", enums.BucketPath.AGGREGATE.value, "aggregate"),
+        ("flat", enums.BucketPath.FLAT.value, "flat"),
+    ]:
+        filenames = transaction.get(filetype, [])
+        for filename in filenames:
+            dp = filename.split("__")[1].split(".")[0]
+            if source == enums.BucketPath.FLAT.value:
```

Comment:
Don't you have some centralized S3 key code to handle these oddities for you?

Reply:
not yet - i can go the other way. I'm going to add this soon, because this path munging is starting to give me a headache.